doc/api/worker_threads.md

```diff
@@ -325,6 +345,30 @@ new Worker('process.env.SET_IN_WORKER = "foo"', { eval: true, env: SHARE_ENV })
   });
 ```
 
+## `worker.setConnectionsListener(fn)`
+
+<!-- YAML
+added: REPLACEME
+-->
+
+> Stability: 1.1 - Active development
+
+* `fn` {Function} A callback to be executed when [`worker.connect()`][] is called from another thread.
+  The function will receive the following arguments:
```

Comment:
Likewise here, an example would be helpful. Also, why wouldn't this be an event on the worker's `process` object? e.g. `process.on('connect', (sender, port, data) => {})`

Reply:
I added a reference to the example above to explain it.

The reason why I haven't added on `process` is because the callback can return a Promise (and therefore be a `AsyncFunction`). As you know we don't want to encourage async listeners in `EE`, isn't it?

Anyway, this API is experimental, in the future I might also add support for this event as fallback.

Reply:
I guess the question I'd have then is does it need to be able to return a promise?

Reply:
+1 for it not returning a promise and being synchronous.

I would still prefer a method over an event since I hate it when we change behavior based on an event being listened to or not (though we do it in several places already)

Reply:
@jasnell Sorry, I don't think I understand the question. `connect` is already `async`, while `setConnectionsListener` explicitly says that `If the function returns a `Promise`, it will be awaited.`, which means that we support `AsyncFunction`.

Anyway, I would support both cases, let me try to provide a `connectSync` function.

Reply:
Edit: since both @jasnell and @benjamingr approved this, I'll wait for this API to settle a little bit and then eventually work on `connectSync` if people ask for it.

Reply:
Note my approval doesn't express "this is done" it's "this looks ready enough for experimental/active development status and for users to test".

The pull request process is tedious for new APIs as is and in my experience if they make sense even if not perfect it's a lot better to land them and iterate than to have a pending PR for a long time.

(James probably means you don't need the promise here since you can just create the MessagePort earlier and buffer its messages until the second side the ready)

Reply:
@benjamingr Sure, I totally get both yours and @jasnell intents. I also do believe we should push this forward to eventually start collecting users' feedback and find an agreement on this aspect of the API :)